application/common/encryption.py

```diff
@@ -0,0 +1,102 @@
+from aws_encryption_sdk import CommitmentPolicy, EncryptionSDKClient, StrictAwsKmsMasterKeyProvider
+from aws_encryption_sdk.exceptions import (
+    MaxEncryptedDataKeysExceeded,
+    NotSupportedError,
+    SerializationError,
+    UnknownIdentityError,
+)
+from aws_lambda_powertools import Logger
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from base64 import b64decode
+from binascii import Error as binasciiError
+from boto3 import client
+from json import loads, dumps
+from os import environ
+from time import time
+from typing import Any, Dict
+
+from common.utilities import extract_body
+
+logger = Logger(child=True)
+
+SIGNING_KEY_KEY = "signing_key"
+BAD_REQUEST = {"statusCode": 400, "body": dumps({"message": "Invalid payload"})}
+
+
+def initialise_encryption_client():
+
+    logger.info("Getting key")
+    kms = client("kms")
+    response = kms.describe_key(
+        KeyId=f"alias/{environ['KEYALIAS']}",
+        GrantTokens=[
+            "string",
+        ],
+    )
+    # Use the key arn you obtained in the previous step
+    key_arn = response["KeyMetadata"]["Arn"]
+    encryption_client = EncryptionSDKClient(commitment_policy=CommitmentPolicy.REQUIRE_ENCRYPT_REQUIRE_DECRYPT)
+
+    # Create an AWS KMS master key provider
+    kms_kwargs = dict(key_ids=[key_arn])
+
+    master_key_provider = StrictAwsKmsMasterKeyProvider(**kms_kwargs)
+
+    def encrypt_string(plaintext):
+
+        encrypted_text, encryptor_header = encryption_client.encrypt(source=plaintext, key_provider=master_key_provider)
+        return encrypted_text
+
+    def decrypt_string(ciphertext):
```

Comment:
I might be missing something but I cannot see where this function is used to decrypt the message on the event sender side. I can see it being used as part the signed checking decorator but that doesn't seem to pass that on to the function

Reply:
then decorator checks the key and then skips the handler returning a 400 if the key is not valid, moves the logic outside of the handler and makes it easy to re-use should we wish to use it on another function, which we wont haha

Reply:
The bit I'm missing how the handler gets the decrypted message when the key is valid, so when I look at it to me I would always expect the sender handler to get an encrypted message

Reply:
lets have a  quick chat after standup